File: C/infix/stack.c
#include "stack.h"
void init(stack *s) {
	s->i = 0;
}
void push(stack *s, int no) {
	s->a[s->i++] = no;
}
int pop(stack *s) {
	int t;
	t = s->a[s->i - 1];
	s->i--;
	return t;
}
int isempty(stack *s) {
	return s->i == 0;
}
int isfull(stack *s) {
	return s->i == MAX;
}

File: stack.js
class Stack {
  constructor () {
    this.s = [];
  }

  peek () {
    return this.s[this.s.length - 1];
  }

  pop () {
    return this.s.pop();
  }

  push (val) {
    this.s.push(val);
    return this.s.length;
  }

  size () {
    return this.s.length;
  }

  isEmpty () {
    return this.s.length === 0;
  }
}

export default Stack
